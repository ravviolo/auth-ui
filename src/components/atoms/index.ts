export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
