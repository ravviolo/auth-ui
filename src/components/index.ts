export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
