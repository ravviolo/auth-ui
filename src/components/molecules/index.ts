export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
