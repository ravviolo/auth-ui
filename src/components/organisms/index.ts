export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
