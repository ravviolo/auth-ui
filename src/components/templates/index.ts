export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
