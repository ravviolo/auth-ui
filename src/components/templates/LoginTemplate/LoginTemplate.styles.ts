import styled from 'styled-components';

import { Flex } from 'components/atoms';

export const Container = styled.main`
  height: 560px;
  padding: 3em;
  background-color: #fff;
  border-radius: 10px;
  box-shadow: 2px 2px 8px 1px rgba(0, 0, 0, 0.3);

  @media screen and (min-width: 1800px) {
    width: 20vw;
  }
  @media screen and (max-width: 1800px) {
    width: 20vw;
  }
  @media screen and (max-width: 1600px) {
    width: 25vw;
  }
  @media screen and (max-width: 1100px) {
    width: 40vw;
  }
  @media screen and (max-width: 700px) {
    width: 50vw;
  }
  @media screen and (max-width: 570px) {
    width: 70vw;
  }
  @media screen and (max-width: 430px) {
    width: 80vw;
  }
`;

export const TemplateFlex = styled(Flex)`
  gap: 2em;
`;
