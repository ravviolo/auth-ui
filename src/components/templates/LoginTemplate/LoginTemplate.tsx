import { Divider } from 'components/atoms';
import { AuthHeader, AuthSocials } from 'components/molecules';
import { LoginForm } from 'components/organisms';

interface Props {
  onSubmit: (
    email: string,
    password: string,
    isPasswordRemembered: boolean
  ) => void;
  onClick: () => void;
  onClickGoogle: () => void;
  onClickFacebook: () => void;
  onClickLinkedIn: () => void;
  onResetPassword: () => void;
}

export const LoginTemplate = ({
  onSubmit,
  onClick,
  onClickFacebook,
  onClickGoogle,
  onClickLinkedIn,
  onResetPassword,
}: Props) => {
  return (
    <div>
      <AuthHeader title="Login" />
      <LoginForm onResetPassword={onResetPassword} onSubmit={onSubmit} />
      <Divider label="OR" />
      <AuthSocials
        onClickFacebook={onClickFacebook}
        onClickGoogle={onClickGoogle}
        onClickLinkedIn={onClickLinkedIn}
      />
    </div>
  );
};
