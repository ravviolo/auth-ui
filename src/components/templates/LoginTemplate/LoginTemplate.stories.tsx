import { ComponentMeta, ComponentStory } from '@storybook/react';

import { LoginTemplate } from 'components/templates';

export default {
  title: 'Templates/LoginTemplate',
  component: LoginTemplate,
  args: {
    footerBtnText: 'Link',
    footerText: 'Footer text.',
    headerText: 'Header Title',
  },
} as ComponentMeta<typeof LoginTemplate>;

export const Default: ComponentStory<typeof LoginTemplate> = (args) => (
  <LoginTemplate {...args} />
);

Default.storyName = 'LoginTemplate';
