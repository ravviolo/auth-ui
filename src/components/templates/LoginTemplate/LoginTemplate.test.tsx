import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { LoginTemplate } from 'components/templates';

import { Props as LoginTemplateProps } from './LoginTemplate';

describe('LoginTemplate', () => {
  const mockOnClickFacebook = jest.fn<
    LoginTemplateProps['onClickFacebook'],
    []
  >();
  const mockOnClickGoogle = jest.fn<LoginTemplateProps['onClickGoogle'], []>();
  const mockOnClickLinkedIn = jest.fn<
    LoginTemplateProps['onClickLinkedIn'],
    []
  >();
  const mockOnSubmitLoginForm = jest.fn<
    LoginTemplateProps['onSubmitLoginForm'],
    [string, string, boolean]
  >();
  const mockOnClickResetPassword = jest.fn<
    LoginTemplateProps['onClickResetPassword'],
    []
  >();
  const mockOnClickFooterBtn = jest.fn<
    LoginTemplateProps['onClickFooterBtn'],
    []
  >();

  const testEmail = 'test@test.com';
  const testPassword = 'testpassword';

  const props: LoginTemplateProps = {
    testId: 'test-login-template',
    headerText: 'Login Header',
    footerText: 'Login Footer',
    footerBtnText: 'Login Footer Button',
    onClickFacebook: mockOnClickFacebook,
    onClickGoogle: mockOnClickGoogle,
    onClickLinkedIn: mockOnClickLinkedIn,
    onSubmitLoginForm: mockOnSubmitLoginForm,
    onClickResetPassword: mockOnClickResetPassword,
    onClickFooterBtn: mockOnClickFooterBtn,
  };

  it('renders login header with title', () => {
    render(<LoginTemplate {...props} />);

    const heading = screen.getByTestId('auth-header-login-template');

    expect(heading).toHaveTextContent('Login Header');
  });

  it('renders login form', () => {
    render(<LoginTemplate {...props} />);

    const loginForm = screen.getByTestId('login-form');

    expect(loginForm).toBeInTheDocument();
  });

  it('submits login form with non-empty input fields', async () => {
    render(<LoginTemplate {...props} />);

    const loginForm = screen.getByTestId('login-form');

    const emailInputField = within(loginForm).getByTestId('input-field-email');
    const passwordInputField = within(loginForm).getByTestId(
      'input-field-password'
    );
    const submitLoginBtn = within(loginForm).getByTestId('btn-submit-login');

    await userEvent.type(emailInputField, testEmail);
    await userEvent.type(passwordInputField, testPassword);
    await userEvent.click(submitLoginBtn);

    expect(mockOnSubmitLoginForm).toHaveBeenCalledTimes(1);
    expect(mockOnSubmitLoginForm).toHaveBeenCalledWith(
      testEmail,
      testPassword,
      false
    );
  });

  it("doesn't submit form with one or more empty input fields", async () => {
    render(<LoginTemplate {...props} />);

    const loginForm = screen.getByTestId('login-form');
    const submitLoginBtn = within(loginForm).getByTestId('btn-submit-login');

    await userEvent.click(submitLoginBtn);

    expect(mockOnSubmitLoginForm).not.toHaveBeenCalled();

    const emailInputField = within(loginForm).getByTestId('input-field-email');
    await userEvent.type(emailInputField, testEmail);
    await userEvent.click(submitLoginBtn);

    expect(mockOnSubmitLoginForm).not.toHaveBeenCalled();
  });

  it("renders a section divider with 'OR' badge", () => {
    render(<LoginTemplate {...props} />);

    const divider = screen.getByTestId('divider-login');

    expect(divider).toHaveTextContent('OR');
  });

  it('renders authenticate through socials section, displays icons for Facebook, Google, LinkedIn', () => {
    render(<LoginTemplate {...props} />);

    const authSocials = screen.getByTestId('auth-socials-login-template');

    const authSocialsIcons = within(authSocials).getAllByTestId(/icon-btn-/);

    expect(authSocialsIcons).toHaveLength(3);
  });

  it('handles click events on Facebook icon - runs handler once', async () => {
    render(<LoginTemplate {...props} />);

    const authSocials = screen.getByTestId('auth-socials-login-template');
    const facebookIcon = within(authSocials).getByTestId(/icon-btn-facebook/i);

    await userEvent.click(facebookIcon);

    expect(mockOnClickFacebook).toHaveBeenCalledTimes(1);
  });

  it('handles click events on Google icon - runs handler once', async () => {
    render(<LoginTemplate {...props} />);

    const authSocials = screen.getByTestId('auth-socials-login-template');
    const googleIcon = within(authSocials).getByTestId(/icon-btn-google/i);

    await userEvent.click(googleIcon);

    expect(mockOnClickGoogle).toHaveBeenCalledTimes(1);
  });

  it('handles click events on LinkedIn icon - runs handler once', async () => {
    render(<LoginTemplate {...props} />);

    const authSocials = screen.getByTestId('auth-socials-login-template');
    const linkedInIcon = within(authSocials).getByTestId(/icon-btn-linkedIn/i);

    await userEvent.click(linkedInIcon);

    expect(mockOnClickLinkedIn).toHaveBeenCalledTimes(1);
  });

  it('renders login footer with text and button', () => {
    render(<LoginTemplate {...props} />);

    const footer = screen.getByTestId('auth-footer-login-template');
    const footerBtn = within(footer).getByTestId('btn-footer');

    expect(footer).toHaveTextContent('Login Footer');
    expect(footerBtn).toHaveTextContent('Login Footer Button');
  });

  it('handles click events on footer button - runs handler once', async () => {
    render(<LoginTemplate {...props} />);

    const footer = screen.getByTestId('auth-footer-login-template');
    const footerBtn = within(footer).getByTestId('btn-footer');

    await userEvent.click(footerBtn);

    expect(mockOnClickFooterBtn).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = create(<LoginTemplate {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
