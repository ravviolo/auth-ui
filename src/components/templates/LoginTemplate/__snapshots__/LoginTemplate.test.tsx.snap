// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginTemplate matches snapshot 1`] = `
<main
  className="sc-dIouRR LRUwC"
  data-testid="test-login-template"
>
  <div
    className="sc-jSMfEi bQrDqy sc-hHLeRK jMMdQ"
    data-testid="flex-login-template"
    direction="column"
  >
    <header
      data-testid="auth-header-login-template"
    >
      <h1
        className="sc-gKXOVf zAjwy"
        data-testid="auth-heading"
      >
        Login Header
      </h1>
    </header>
    <form
      data-testid="login-form"
      onSubmit={[Function]}
    >
      <div
        className="sc-jSMfEi bQrDqy sc-hAZoDl bWsHpL"
        data-testid="flex-login-form"
        direction="column"
      >
        <div
          className="sc-jSMfEi bQrDqy sc-breuTD jjaaRC"
          data-testid="email-input-login-form"
          direction="column"
        >
          <label
            className="sc-papXJ daxRFt"
            data-testid="input-label-email"
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="sc-ftvSup dLJLzu"
            data-testid="input-field-email"
            id="email"
            name="email"
            type="email"
          />
        </div>
        <div
          className="sc-jSMfEi bQrDqy sc-fnykZs hELmKH"
          data-testid="flex-login-form-password-control"
          direction="column"
        >
          <div
            className="sc-jSMfEi bQrDqy sc-ksZaOG cRa-dNN"
            data-testid="password-input-login-form"
            direction="column"
          >
            <label
              className="sc-papXJ daxRFt"
              data-testid="input-label-password"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="sc-ftvSup dLJLzu"
              data-testid="input-field-password"
              id="password"
              name="password"
              type="password"
            />
          </div>
          <div
            className="sc-jSMfEi iChCMC sc-fEOsli dPSjkT"
            data-testid="flex-login-form-checkbox"
            direction="row"
          >
            <input
              className="sc-gsnTZi kVuJZl"
              data-testid="remember-password-checkbox"
              id="rememberPassword"
              name="rememberPassword"
              type="checkbox"
            />
            <label
              className="sc-papXJ daxRFt"
              data-testid="input-label-remember-password-checkbox"
              htmlFor="rememberPassword"
            >
              Remember me?
            </label>
          </div>
        </div>
        <button
          className="sc-bczRLJ jXOWFG"
          data-testid="btn-submit-login"
          type="submit"
        >
          Login
        </button>
        <button
          className="sc-jqUVSM ebpvV sc-bjUoiL hqHESO"
          data-testid="btn-reset-password"
          onClick={[MockFunction]}
          type="button"
        >
          Forgot Password?
        </button>
      </div>
    </form>
    <div
      className="sc-dkzDqf fBSlBa"
      data-testid="divider-login"
    >
      <div
        className="sc-hKMtZM cJojKy"
      />
      <span
        className="sc-eCYdqJ fGmLST"
      >
        OR
      </span>
    </div>
    <div
      className="sc-jSMfEi iChCMC sc-evZas fBPmfS"
      data-testid="auth-socials-login-template"
      direction="row"
    >
      <button
        className="sc-iBkjds cxbDaZ"
        color="#DB4437"
        data-testid="icon-btn-Google"
        onClick={[MockFunction]}
        type="button"
      >
        G
      </button>
      <button
        className="sc-iBkjds bqhZzF"
        color="#4267B2"
        data-testid="icon-btn-Facebook"
        onClick={[MockFunction]}
        type="button"
      >
        f
      </button>
      <button
        className="sc-iBkjds hAqsSt"
        color="#0077B5"
        data-testid="icon-btn-LinkedIn"
        onClick={[MockFunction]}
        type="button"
      >
        in
      </button>
    </div>
    <footer
      className="sc-kDDrLX kktpMP"
      data-testid="auth-footer-login-template"
    >
      <div
        className="sc-jSMfEi iChCMC sc-crXcEl bUmuaF"
        data-testid="flex-footer"
        direction="row"
      >
        <p>
          Login Footer
        </p>
        <button
          className="sc-jqUVSM ebpvV sc-iqcoie bPYbbt"
          data-testid="btn-footer"
          onClick={[MockFunction]}
          type="button"
        >
          Login Footer Button
        </button>
      </div>
    </footer>
  </div>
</main>
`;
